MobileMink immediately crashes when used on Android 4.4.4

[Screencast](https://youtu.be/07NpSpL1XMA) of user experience on Android 4.4.4 with April 15, 2015 APK despite fix in #1.

@Thing342 @jbrunelle

Verified the same behavior with Mobile Mink 0.5.1 on 201504301543 on 4.4.4 device.

Some dump info (from the emulator's debugger):

``` sh
04-30 18:33:35.006    1010-1289/org.mitre.mobilememento E/AndroidRuntime﹕ FATAL EXCEPTION: AsyncTask #1
    Process: org.mitre.mobilememento, PID: 1010
    java.lang.RuntimeException: An error occured while executing doInBackground()
            at android.os.AsyncTask$3.done(AsyncTask.java:300)
            at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
            at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
            at java.util.concurrent.FutureTask.run(FutureTask.java:242)
            at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
            at java.lang.Thread.run(Thread.java:841)
     Caused by: java.lang.NullPointerException
            at org.mitre.mobilememento.TimeMap.<init>(TimeMap.java:33)
            at org.mitre.mobilememento.TimeMap.union(TimeMap.java:54)
            at org.mitre.mobilememento.ViewArchiveActivity$GenerateTimeMapTask.doInBackground(ViewArchiveActivity.java:266)
            at org.mitre.mobilememento.ViewArchiveActivity$GenerateTimeMapTask.doInBackground(ViewArchiveActivity.java:224)
            at android.os.AsyncTask$2.call(AsyncTask.java:288)
            at java.util.concurrent.FutureTask.run(FutureTask.java:237)
```

and 

``` sh
04-30 18:33:37.666    1010-1010/org.mitre.mobilememento E/WindowManager﹕ android.view.WindowLeaked: Activity org.mitre.mobilememento.ViewArchiveActivity has leaked window com.android.internal.policy.impl.PhoneWindow$DecorView{b1e54dc0 V.E..... R......D 0,0-1026,486} that was originally added here
            at android.view.ViewRootImpl.<init>(ViewRootImpl.java:348)
            at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:248)
            at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:69)
            at android.app.Dialog.show(Dialog.java:286)
            at android.app.ProgressDialog.show(ProgressDialog.java:116)
            at android.app.ProgressDialog.show(ProgressDialog.java:99)
            at android.app.ProgressDialog.show(ProgressDialog.java:94)
            at org.mitre.mobilememento.ViewArchiveActivity$GenerateTimeMapTask.onPreExecute(ViewArchiveActivity.java:234)
            at android.os.AsyncTask.executeOnExecutor(AsyncTask.java:587)
            at android.os.AsyncTask.execute(AsyncTask.java:535)
            at org.mitre.mobilememento.ViewArchiveActivity.onCreate(ViewArchiveActivity.java:70)
            at android.app.Activity.performCreate(Activity.java:5231)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2159)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2245)
            at android.app.ActivityThread.access$800(ActivityThread.java:135)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1196)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:136)
            at android.app.ActivityThread.main(ActivityThread.java:5017)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:515)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
            at dalvik.system.NativeStart.main(Native Method)
```

from Android 4.4.2

--- a/app/src/main/java/org/mitre/mobilememento/TimeMap.java
+++ b/app/src/main/java/org/mitre/mobilememento/TimeMap.java
@@ -30,7 +30,7 @@ public class TimeMap implements Comparator<Memento> {
     private TimeMap(TimeMap... others) {
         this.screenType = ScreenType.DESKTOP;
         if (others.length != 0) {
-            this.contentUrl = others[0].contentUrl;
+            if(others[0] != null)   this.contentUrl = others[0].contentUrl; // Prevent accessor to invalid TM object
             for (TimeMap map : others) if (map != null) mementos.addAll(map.getMementos());
         }
     }

